General Error Deleting DAG run

Using a 2.0.0 beta 2 build...

![Untitled](https://user-images.githubusercontent.com/4283/99546252-3ebe1a80-2984-11eb-9681-8d6e718b22c5.gif)

Log from webserver:

```
[2020-11-18 14:58:52,712] {interface.py:713} ERROR - Delete record error: Dependency rule tried to blank-out primary key column 'job.id' on instance '<SchedulerJob at 0x7fc68ba7f990>'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/flask_appbuilder/models/sqla/interface.py", line 698, in delete
    self.session.commit()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/scoping.py", line 163, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 1042, in commit
    self.transaction.commit()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 504, in commit
    self._prepare_impl()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 483, in _prepare_impl
    self.session.flush()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2536, in flush
    self._flush(objects)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2678, in _flush
    transaction.rollback(_capture_exception=True)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    with_traceback=exc_tb,
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2638, in _flush
    flush_context.execute()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 538, in execute
    self.dependency_processor.process_deletes(uow, states)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/dependency.py", line 547, in process_deletes
    state, child, None, True, uowcommit, False
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/dependency.py", line 604, in _synchronize
    sync.clear(dest, self.mapper, self.prop.synchronize_pairs)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/sync.py", line 88, in clear
    "column '%s' on instance '%s'" % (r, orm_util.state_str(dest))
AssertionError: Dependency rule tried to blank-out primary key column 'job.id' on instance '<SchedulerJob at 0x7fc68ba7f990>'
172.19.0.1 - - [18/Nov/2020:14:58:52 +0000] "POST /dagrun/delete/147 HTTP/1.1" 302 415 "http://localhost:8080/dagrun/list/?_flt_3_dag_id=airflow_community_data_refresh&_flt_3_state=success" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.193 Safari/537.36"
```

--- a/airflow/models/serialized_dag.py
+++ b/airflow/models/serialized_dag.py
@@ -25,7 +25,7 @@ from typing import Any, Dict, List, Optional
 
 import sqlalchemy_jsonfield
 from sqlalchemy import BigInteger, Column, Index, String, and_
-from sqlalchemy.orm import Session, backref, relationship, foreign
+from sqlalchemy.orm import Session, backref, foreign, relationship
 from sqlalchemy.sql import exists
 
 from airflow.models.base import ID_LEN, Base

--- a/airflow/models/serialized_dag.py
+++ b/airflow/models/serialized_dag.py
@@ -25,7 +25,7 @@ from typing import Any, Dict, List, Optional
 
 import sqlalchemy_jsonfield
 from sqlalchemy import BigInteger, Column, Index, String, and_
-from sqlalchemy.orm import Session, backref, relationship
+from sqlalchemy.orm import Session, backref, relationship, foreign
 from sqlalchemy.sql import exists
 
 from airflow.models.base import ID_LEN, Base
@@ -73,7 +73,7 @@ class SerializedDagModel(Base):
 
     dag_runs = relationship(
         DagRun,
-        primaryjoin=dag_id == DagRun.dag_id,
+        primaryjoin=dag_id == foreign(DagRun.dag_id),
         backref=backref('serialized_dag', uselist=False, innerjoin=True),
     )
 

--- a/airflow/models/serialized_dag.py
+++ b/airflow/models/serialized_dag.py
@@ -25,7 +25,7 @@ from typing import Any, Dict, List, Optional
 
 import sqlalchemy_jsonfield
 from sqlalchemy import BigInteger, Column, Index, String, and_
-from sqlalchemy.orm import Session, backref, relationship
+from sqlalchemy.orm import Session, backref, relationship, foreign
 from sqlalchemy.sql import exists
 
 from airflow.models.base import ID_LEN, Base
@@ -73,8 +73,7 @@ class SerializedDagModel(Base):
 
     dag_runs = relationship(
         DagRun,
-        primaryjoin=dag_id == DagRun.dag_id,
-        foreign_keys=dag_id,
+        primaryjoin=dag_id == foreign(DagRun.dag_id),
         backref=backref('serialized_dag', uselist=False, innerjoin=True),
     )
 

--- a/airflow/models/serialized_dag.py
+++ b/airflow/models/serialized_dag.py
@@ -25,7 +25,7 @@ from typing import Any, Dict, List, Optional
 
 import sqlalchemy_jsonfield
 from sqlalchemy import BigInteger, Column, Index, String, and_
-from sqlalchemy.orm import Session, backref, relationship, foreign
+from sqlalchemy.orm import Session, backref, relationship
 from sqlalchemy.sql import exists
 
 from airflow.models.base import ID_LEN, Base
@@ -73,7 +73,7 @@ class SerializedDagModel(Base):
 
     dag_runs = relationship(
         DagRun,
-        primaryjoin=dag_id == foreign(DagRun.dag_id),
+        primaryjoin=dag_id == DagRun.dag_id,
         backref=backref('serialized_dag', uselist=False, innerjoin=True),
     )
 

--- a/tests/models/test_dagrun.py
+++ b/tests/models/test_dagrun.py
@@ -788,3 +788,12 @@ class TestDagRun(unittest.TestCase):
         ti_failed = dag_run.get_task_instance(dag_task_failed.task_id)
         self.assertIn(ti_success.state, State.success_states)
         self.assertIn(ti_failed.state, State.failed_states)
+
+    def test_delete_dag_run(self):
+        dag = DAG(dag_id='test_delete_dag_run', start_date=days_ago(1))
+
+        dag_run = self.create_dag_run(dag=dag)
+
+        session = settings.Session()
+        session.delete(dag_run)
+        session.commit()

